### PR TITLE
Tidy the Gradle build files

### PR DESCRIPTION
**Pull request by 3flex**
Tidies up the Gradle build files, using some of the newer syntax from Gradle Kotlin DSL, a review of some of the module dependencies, and removing some of the things added with the introduction of the composite build that I'm now aware are not required (though the composite build remains).

Less code = Less bugs (hopefully!)

### CODE REVIEW

**Comment by pavlospt** [comment]
Nice work!

**Comment by 3flex** [comment]
Looks like an issue pulling a dependency on AppVeyor causing a build failure. Travis and one of the AppVeyor builds ran OK.

**Comment by arturbosch** [comment]
Thanks for the clean up, please rebase the PR and I will gladly merge this!

**Comment by 3flex** [comment]
Rebased, and made a couple more minor tweaks.

File: detekt-core/build.gradle.kts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 val junitPlatformVersion: String by project
 val spekVersion: String by project
 
 dependencies {
+	implementation(kotlin("compiler-embeddable"))
 	api(project(":detekt-api"))
 
 	testImplementation(project(":detekt-rules"))
 	testImplementation(project(":detekt-test"))
+	testImplementation("org.reflections:reflections:0.9.11")
```
Comment (arturbosch) [inline detekt-core/build.gradle.kts]: Please extract this version.

File: detekt-cli/build.gradle.kts [hunk 2]
```diff
@@ -13,20 +12,17 @@ configurations.testImplementation.extendsFrom(configurations.kotlinTest)
...
 dependencies {
 	implementation(project(":detekt-core"))
-	implementation(project(":detekt-rules"))
 	implementation("com.beust:jcommander:$jcommanderVersion")
-	implementation("org.jetbrains.kotlin:kotlin-compiler-embeddable:$kotlinVersion")
+	implementation(kotlin("compiler-embeddable"))
 
 	testImplementation(project(":detekt-test"))
+	testImplementation(project(":detekt-rules"))
+	testImplementation("org.reflections:reflections:0.9.11")
```
Comment (arturbosch) [inline detekt-cli/build.gradle.kts]: Please extract this version.

File: build.gradle.kts [hunk 3]
```diff
@@ -1,34 +1,18 @@
...
-	val kotlinVersion: String by project
-
-	dependencies {
-		classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
-	}
-}
+import java.util.Date
 
 plugins {
+	kotlin("jvm") version "1.2.61"
```
Comment (arturbosch) [inline build.gradle.kts]: Hm I think having the kotlin version here is okay as I use the dependency update plugin.

File: detekt-rules/build.gradle.kts [hunk 4]
```diff
@@ -1,18 +1,15 @@
...
+tasks["build"].finalizedBy(":detekt-generator:generateDocumentation")
 
 val junitPlatformVersion: String by project
 val spekVersion: String by project
 
 dependencies {
 	implementation(project(":detekt-api"))
+	implementation(kotlin("compiler-embeddable"))
 
+	testImplementation("org.reflections:reflections:0.9.11")
```
Comment (arturbosch) [inline detekt-rules/build.gradle.kts]: Here third time the version :).

File: detekt-test/build.gradle.kts [hunk 5]
```diff
@@ -1,15 +1,7 @@
-configurations.implementation.extendsFrom(configurations.kotlinImplementation)
-configurations.testImplementation.extendsFrom(configurations.kotlinTest)
-
 val assertjVersion: String by project
-val junitPlatformVersion: String by project
-val spekVersion: String by project
 
 dependencies {
-	implementation(project(":detekt-rules"))
```
Comment (arturbosch) [inline detekt-test/build.gradle.kts]: Hmm, why test depended on rules ? And is this not necessary anymore?

Comment (3flex) [reply detekt-test/build.gradle.kts]: If a module builds without an implementation or api dependency, then by definition the dependency isn't required.

I didn't look at the commit history. But there are currently no `io.gitlab.arturbosch.detekt.rules.*` imports in detekt-test, so it's not being used there today.

**Comment by 3flex** [comment]
Build failure on AppVeyor due to a network timeout.

**Pull request merged**